tests/unit/testaddon.cpp

```diff
@@ -1147,11 +1147,11 @@ void TestAddon::telemetry_state_change() {
   // Event metrics can only record InvalidValue or InvalidOverflow errors
   // https://mozilla.github.io/glean/book/reference/metrics/event.html#recorded-errors
 
-  QCOMPARE(mozilla::glean::sample::addon_state_changed.testGetNumRecordedErrors(
-               VPNGlean::InvalidValue),
+  QCOMPARE(mozilla::glean::sample::addon_state_changed.numRecordedErrors(
+               EventMetric::InvalidValue),
```

Comment:
The ErrorType is not metric specific. It should remain in the VPNGlean class.

Reply:
I disagree, the `ErrorType` is only used as a part of the `EventMetric` class, so it should remain as a part of this library. Moving it back into `VPNGlean` would introduce circular dependencies between `vpnglean` and the VPN app.

Reply:
If we think this enum needs to be moved out of the `EventMetric` class, I would create a new header for it in the `vpnglean` directory.

Reply:
This _will_ be moved out of the EventMetric class. Glean has multiple metric types. Although we currently only implement the Event metric, the plan is to implement more as we move on. All metric types share the ErrorType enum. As the comment says, we don't need to define this enum at all and instead import it through the FFI. Let's just leave it like this for now an work on https://mozilla-hub.atlassian.net/browse/VPN-3339 after.